### PR TITLE
feat: use new beabee-common functionality

### PR DESCRIPTION
**Pull request by wpf500**
Use the new beabee-common library with lots of improvements to types and some validators

### CODE REVIEW

File: src/api/validators/IsPassword.ts [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
-  validate(password: unknown): boolean | Promise<boolean> {
-    return (
-      typeof password === "string" &&
-      password.length >= 8 &&
-      /[a-z]/.test(password) &&
-      /[A-Z]/.test(password) &&
-      /[0-9]/.test(password)
-    );
-  }
+export function isPassword(password: unknown): boolean {
```
Comment (JumpLink) [inline src/api/validators/IsPassword.ts:3]: Would make sense to move this method to `beabee-common` to my validation branch

File: src/api/validators/IsSlug.ts [hunk 2]
```diff
@@ -1,16 +1,18 @@
 import { buildMessage, ValidateBy, ValidationOptions } from "class-validator";
 import slugify from "slugify";
 
+export function isSlug(slug: unknown): boolean {
```
Comment (JumpLink) [inline src/api/validators/IsSlug.ts:4]: Can also be moved to `beabee-common@callout-validation`.

File: src/api/validators/IsType.ts [hunk 3]
```diff
@@ -1,7 +1,21 @@
 import { ValidateBy, ValidationOptions } from "class-validator";
 import { ValidationArguments } from "class-validator/types/validation/ValidationArguments";
 
-const IS_TYPE = "isType";
+export function isType(
```
Comment (JumpLink) [inline src/api/validators/IsType.ts:4]: Can also be moved to `beabee-common@callout-validation`.

**Review by JumpLink — COMMENTED** [review]
Looks very good and clean to me, but I wonder why we don't outsource the validation directly to common. If we don`t want to have `class-validator` as a dependency in common, we can only move the small helper functions, otherwise maybe more parts of the validation could be moved to common.

**Comment by wpf500** [comment]
Moved validators to https://github.com/beabee-communityrm/beabee-common/pull/8, will update this PR once a new beabee-common has been merged

**Comment by JumpLink** [comment]
@wpf500 What do we need to make this PR ready? I've described [here](https://github.com/beabee-communityrm/beabee/tree/feat/payment-paypal?tab=readme-ov-file#-common-package) how you can test any changes on beabee-common locally on the backend app without the need to publish them on NPM, which should make it much easier to work on it.

**Comment by wpf500** [comment]
@JumpLink this is ready for your review. When https://github.com/beabee-communityrm/beabee-common/pull/15 has been merged I will release beabee-common@v0.20.2 so this doesn't reference an alpha package

File: src/api/dto/CalloutFormDto.ts [hunk 4]
```diff
@@ -72,57 +55,84 @@ abstract class BaseCalloutComponentDto implements BaseCalloutComponentSchema {
...
+  @IsOptional()
+  @IsObject()
+  validate?: CalloutComponentBaseRules;
+
+  @IsOptional()
+  @IsBoolean()
+  multiple?: boolean;
+
+  @IsOptional()
+  defaultValue?: CalloutResponseAnswer | CalloutResponseAnswer[] | null;
```
Comment (JumpLink) [inline src/api/dto/CalloutFormDto.ts:88]: content components have no inputs and only output html, we still need the properties `validate`, `multiple` and `defaultValue`?

File: src/api/dto/CalloutFormDto.ts [hunk 5]
```diff
@@ -157,20 +164,24 @@ function ComponentType() {
...
-        default:
-          throw new Error("Unknown component type " + component.type);
+      switch (component.type) {
+        case CalloutComponentType.CONTENT:
+          return plainToInstance(CalloutComponentContentDto, component);
+        case CalloutComponentType.INPUT_SELECT:
+          return plainToInstance(CalloutComponentInputSelectDto, component);
+      }
+
+      if (calloutComponentInputSelectableTypes.includes(component.type)) {
```
Comment (JumpLink) [inline src/api/dto/CalloutFormDto.ts:174]: You could also use `isCalloutComponentOfBaseType(component, CalloutComponentBaseType.NESTABLE)` from the common package here. Same for the base types bellow

**Review by JumpLink — APPROVED** [review]
Looks good to me, just left a few comments again

File: src/api/dto/CalloutFormDto.ts (hunk 4 above)
Comment (wpf500) [reply src/api/dto/CalloutFormDto.ts:88]: I just implemented the types that come from `CalloutComponentBaseSchema`, so I think these probably aren't necesssary but that would need to be updated in beabee-common

File: src/api/dto/CalloutFormDto.ts (hunk 5 above)
Comment (wpf500) [reply src/api/dto/CalloutFormDto.ts:174]: Woooo :tada:

File: src/api/dto/CalloutFormDto.ts (hunk 4 above)
Comment (JumpLink) [reply src/api/dto/CalloutFormDto.ts:90]: Than you could use the `CalloutComponentBaseInputSchema` interface for that

**Pull request merged**